impeller/renderer/backend/vulkan/fence_waiter_vk.cc

```diff
@@ -54,56 +53,36 @@ void FenceWaiterVK::Main() {
     wait_set_cv_.wait(lock, [&]() { return !wait_set_.empty() || terminate_; });
 
     auto device_holder = device_holder_.lock();
-    if (!device_holder) {
+    if (!device_holder || terminate_) {
       break;
     }
 
-    auto wait_set = TrimAndCreateWaitSetLocked(device_holder);
-
+    std::vector<vk::UniqueFence> temp_wait_set = std::move(wait_set_);
+    std::vector<fml::closure> callbacks = std::move(wait_set_callbacks_);
     lock.unlock();
 
-    if (!wait_set.has_value()) {
-      break;
+    std::vector<vk::Fence> fences(temp_wait_set.size());
+    for (auto i = 0u; i < temp_wait_set.size(); i++) {
+      fences[i] = temp_wait_set[i].get();
     }
-    if (wait_set->empty()) {
+
+    if (fences.empty()) {
       continue;
     }
 
     auto result = device_holder->GetDevice().waitForFences(
-        wait_set->size(),                        // fences count
-        wait_set->data(),                        // fences
-        false,                                   // wait for all
+        fences.size(),                           // fences count
+        fences.data(),                           // fences
+        true,                                    // wait for all
         std::chrono::nanoseconds{100ms}.count()  // timeout (ns)
```

Comment:
Should we adjust the timeout now that we are waiting on all of them?  100ms / fence?

Reply:
i'd defer to @chinmaygarde on that, I left the old default but all the values seem pretty arbitrary. I guess if we're waiting for all of them,it makes sense to make it longer but not sure how long.

Reply:
maybe if we hit the timeout condition, I grab the mutex again and add all the fences back into the set?